packages/kotti-ui/source/kotti-field-select/components/ActionIcon.vue

```diff
@@ -1,52 +1,64 @@
 <template>
-	<div
-		:class="classes"
-		role="button"
-		@mouseenter="hoverOnClearIcon = true"
-		@mouseleave="hoverOnClearIcon = false"
-	>
-		<i class="yoco" @click.stop="handleClearIfShown" v-text="icon" />
+	<div :class="classes" role="button">
+		<div class="action-icon__container">
+			<i
+				v-if="canClear"
+				class="yoco"
+				@click="handleClearIfShown"
+				v-text="Yoco.Icon.CLOSE"
+			/>
+			<i
+				class="yoco"
+				@click.stop.prevent="handleSetIsDropdownOpen"
+				@mousedown.stop.prevent="() => {}"
+				v-text="isDropdownOpen ? Yoco.Icon.CHEVRON_UP : Yoco.Icon.CHEVRON_DOWN"
+			/>
+		</div>
 	</div>
 </template>
 
 <script lang="ts">
 import { Yoco } from '@3yourmind/yoco'
-import { defineComponent, ref, computed } from '@vue/composition-api'
+import { defineComponent, computed } from '@vue/composition-api'
 
 export default defineComponent({
 	name: 'ActionIcon',
 	props: {
 		classes: { required: true, type: Array },
 		handleClear: { required: true, type: Function },
 		isDropdownOpen: { required: true, type: Boolean },
+		isFieldHoveredOrFocused: { required: true, type: Boolean },
 		showClear: { required: true, type: Boolean },
 	},
-	setup(props: {
-		classes: string[]
-		handleClear(): void
-		isDropdownOpen: boolean
-		showClear: boolean
-	}) {
-		const hoverOnClearIcon = ref(false)
-
+	setup(
+		props: {
+			classes: string[]
+			handleClear(): void
+			isFieldHoveredOrFocused: boolean
+			isDropdownOpen: boolean
+			showClear: boolean
+		},
+		{ emit },
+	) {
 		const canClear = computed<boolean>(
-			() => props.showClear && hoverOnClearIcon.value,
+			() => props.showClear && props.isFieldHoveredOrFocused,
 		)
 
 		return {
+			canClear,
 			handleClearIfShown: () => {
 				if (canClear.value) props.handleClear()
 			},
-			hoverOnClearIcon,
-			icon: computed(
-				(): Yoco.Icon =>
-					canClear.value
-						? Yoco.Icon.CLOSE
-						: props.isDropdownOpen
-						? Yoco.Icon.CHEVRON_UP
-						: Yoco.Icon.CHEVRON_DOWN,
-			),
+			handleSetIsDropdownOpen: () =>
+				emit('setIsDropdownOpen', !props.isDropdownOpen),
+			Yoco,
 		}
 	},
 })
 </script>
+
+<style lang="scss" scoped>
+.action-icon__container {
+	display: flex;
```

Comment:
don't you need `justify-content: center`

i think the default would be `flex-start` here which would left align the icons instead of center them in the space given

Reply:
Probably your solution its more correct, however the`action-icon__container` div's width is equal to the width of the icons that. are inside of it thats why it doesn't have any consequences adding it or not, but I will add it just in case that changes in the future.